packages/compiler/src/server/completion.ts

```diff
@@ -436,6 +442,26 @@ function addIdentifierCompletion(
       kind,
       insertText: printIdentifier(key) + (suffix ?? ""),
     };
+    if (sym.name.startsWith("$")) {
+      const targetNode = getSourceLocation(node);
+      const lineAndChar = targetNode.file.getLineAndCharacterOfPosition(node.pos);
+      item = {
+        label: label ?? key,
```

Comment:
can you not patch the existing item object instead of redefining the common pieces?